MetaMask/metamask-extension
# feat: funding method modal

## **Description**

Developed the "FundingMethodModal". This CTA displays when a user has no native token balance, and triggering the CTA will now render a modal with 3 options:

1. "Buy" - Navigates user to Portfolio Buy page
2. "Receive" - Renders the QR code modal inside the extension
3. "Transfer" - Will navigate the user to the "Transfers" page with their associated account address to be consumed inside the Portfolio app for CEX transfers

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26426?quickstart=1)

## **Related issues**

Fixes: --

## **Manual testing steps**

1. Inside extension, select an account with a 0 token balance
2. Click the "Get Started" CTA
3. See that the modal renders and looks good in both light + dark mode
4. Click on all 3 of the funding method options and ensure the appropriate action is triggered

**NOTE**

We currently have a feature branch in Portfolio for "Transfer", so this page (`/transfer`) is not currently in prod. Clicking the "Transfer funds" CTA inside the FundingMethodsModal will give you a page not found error at the moment. This PR will get merge once that code, in Portfolio, is live.

## **Screenshots/Recordings**

### **Before**
<img width="353" alt="Screenshot 2024-08-14 at 12 21 59 PM" src="https://github.com/user-attachments/assets/6240f3a8-48bb-4363-8686-e1c50b6fad21">

### **After**
<img width="354" alt="Screenshot 2024-08-14 at 10 59 15 AM" src="https://github.com/user-attachments/assets/8dbdf44b-7794-4fbf-b05b-51fbeca58a0b">
<img width="359" alt="Screenshot 2024-08-14 at 10 59 25 AM" src="https://github.com/user-attachments/assets/2356f299-acc7-4d1c-b809-53ffac8d47c6">
<img width="356" alt="Screenshot 2024-08-14 at 10 59 31 AM" src="https://github.com/user-attachments/assets/303d38c1-bb04-48d7-b1b1-c72687305b0c">
<img width="357" alt="Screenshot 2024-08-14 at 10 59 58 AM" src="https://github.com/user-attachments/assets/9591e55e-b1be-4fac-b322-25c7b791b5df">

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
